app/services/generate_application_ref.rb

```diff
@@ -0,0 +1,16 @@
+class GenerateApplicationRef
+  REFERENCE_LENGTH = 6
+
+  class << self
+    def call(application_form)
+      application_form.update_attribute(:reference, new_application_reference)
```

Comment:
Instead of making the service update a provided `application_form`, we could have it return the reference and then we can assign it ourselves